test/deployer/deployer.js

```diff
@@ -206,5 +219,34 @@ describe('Deployer tests', () => {
 
 			assert.equal(gas, estimateGas.toString())
 		})
+
+		it('should wrap contracts correctly', async () => {
+			const gas = 1629070;
+
+			let libraries = {"LinkedList": "0x2Be52D5d7A73FC183cF40053B95beD572519EBbC"};
+			const estimateGas = await deployer.estimateGas(DataContract, libraries);
+
+			assert.equal(gas, estimateGas.toString())
+		})
+	})
+
+	describe('Preparing bytecode', async () => {
+		let wallet
+		let provider;
+		let deployer;
+
+		beforeEach(async () => {
+			wallet = new ethers.Wallet('0x' + config.randomPrivateKey);
+			provider = new ethers.providers.JsonRpcProvider(config.nodeUrl, ethers.providers.networks.unspecified);
+			deployer = new etherlime.Deployer(wallet, provider, defaultConfigs);
+
+		})
+
+		it('should prepare the bytecode', async () => {
+			let invalidLibrary = 100;
+			let bytecode = await deployer._prepareBytecode(invalidLibrary, ICOTokenContract.bytecode);
```

Comment:
How can a test with invalid library be correct test? Or maybe the it message is misleading?

Reply:
If the library is invalid it should not link it to the bytecode. This means that when _prepareBytecode is executed with an invalid library the returned bytecode should be the same without any changes. I'll update the message